PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ["hr.applicant"]
-    _inherit = ["hr.job"]
-    _inherit = ['survey.survey']
-    _inherit = ["survey.user_input"]

Total Changes: 8

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_recruitment_survey/models/hr_applicant.py
+++ b/addons/hr_recruitment_survey/models/hr_applicant.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class HrApplicant(models.Model):
-    _inherit = ["hr.applicant"]
+    _inherit = "hr.applicant"
 
     survey_id = fields.Many2one('survey.survey', related='job_id.survey_id', string="Survey", readonly=True)
     response_ids = fields.One2many('survey.user_input', 'applicant_id', string="Responses")

--- a/addons/hr_recruitment_survey/models/hr_job.py
+++ b/addons/hr_recruitment_survey/models/hr_job.py
@@ -4,7 +4,7 @@ from odoo import fields, models, _
 
 
 class HrJob(models.Model):
-    _inherit = ["hr.job"]
+    _inherit = "hr.job"
 
     survey_id = fields.Many2one(
         'survey.survey', "Interview Form",

--- a/addons/hr_recruitment_survey/models/survey_survey.py
+++ b/addons/hr_recruitment_survey/models/survey_survey.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class SurveySurvey(models.Model):
-    _inherit = ['survey.survey']
+    _inherit = 'survey.survey'
 
     survey_type = fields.Selection(selection_add=[('recruitment', 'Recruitment')], ondelete={'recruitment': 'set default'})
     hr_job_ids = fields.One2many("hr.job", "survey_id", string="Job Position")

--- a/addons/hr_recruitment_survey/models/survey_user_input.py
+++ b/addons/hr_recruitment_survey/models/survey_user_input.py
@@ -4,7 +4,7 @@ from odoo import fields, models, _
 
 
 class SurveyUser_Input(models.Model):
-    _inherit = ["survey.user_input"]
+    _inherit = "survey.user_input"
 
     applicant_id = fields.Many2one('hr.applicant', string='Applicant', index='btree_not_null')
 
